### ISSUE TITLE
`no-self-assign` rule failing for shorthand assignment operators

### ISSUE THREAD

**Issue by pillowfication** [eslint/eslint#5371]
```javascript
/*eslint no-self-assign: 2*/
var a = 5;
a += a;        // error 'a' is assigned to itself
```
This fails for all the [similar operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Assignment_Operators) when it should be allowed.

**Comment by mysticatea** [comment]
Thank you for this issue.

It's my mistake!
